File: user-db/src/main/java/group_05/ase/user_db/restData/InterestDTO.java
package group_05.ase.user_db.restData;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class InterestDTO {

    @Getter
    @JsonProperty(required = true, value = "interest_id")
    private int interestId;

    @Getter
    @Setter
    @JsonProperty(required = true, value = "interest_name")
    private String interestNameEn;

    @Getter
    @Setter
    @JsonProperty(required = true, value = "description")
    private String description;

    @Getter
    @Setter
    @JsonProperty(required = true, value = "interest_name_de")
    private String interestNameDe;

    @Override
    public String toString() {
        return "InterestDTO{" +
                "interestId=" + interestId +
                ", interestNameEn='" + interestNameEn + '\'' +
                ", description='" + description + '\'' +
                ", interestNameDe='" + interestNameDe + '\'' +
                '}';
    }

}

File: frontend/src/app/dto/interest.dto.ts
export class InterestDto {
  protected interestId: number;
  private interestName: string;
  private description: string;

  constructor( interestId: number, interestName: string, description: string) {
    this.interestId = interestId;
    this.interestName = interestName;
    this.description = description;
  }

  getInterestId(): number {
    return this.interestId;
  }

  getInterestName(): string {
    return this.interestName;
  }

  getDescription(): string {
    return this.description;
  }

}
